homeassistant/components/device_tracker/automatic.py

```diff
@@ -52,64 +60,172 @@ def async_setup_scanner(hass, config, async_see, discovery_info=None):
     try:
         session = yield from client.create_session_from_password(
             SCOPE, config[CONF_USERNAME], config[CONF_PASSWORD])
-        data = AutomaticData(hass, session, config[CONF_DEVICES], async_see)
+        data = AutomaticData(
+            hass, client, session, config[CONF_DEVICES], async_see)
+
+        # Load the initial vehicle data
+        vehicles = yield from session.get_vehicles()
+        for vehicle in vehicles:
+            hass.async_add_job(data.load_vehicle(vehicle))
     except aioautomatic.exceptions.AutomaticError as err:
         _LOGGER.error(str(err))
         return False
 
-    yield from data.update()
+    @callback
+    def ws_connect(event):
+        """Open the websocket connection."""
+        hass.async_add_job(data.ws_connect())
+
+    @callback
+    def ws_close(event):
+        """Close the websocket connection."""
+        hass.async_add_job(data.ws_close())
+
+    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_START, ws_connect)
+    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, ws_close)
+
     return True
 
 
 class AutomaticData(object):
     """A class representing an Automatic cloud service connection."""
 
-    def __init__(self, hass, session, devices, async_see):
+    def __init__(self, hass, client, session, devices, async_see):
         """Initialize the automatic device scanner."""
         self.hass = hass
         self.devices = devices
+        self.vehicle_info = {}
+        self.client = client
         self.session = session
         self.async_see = async_see
+        self.ws_reconnect_handle = None
+        self.ws_close_requested = False
+
+        self.client.on_app_event(
+            lambda name, event: self.hass.async_add_job(
+                self.handle_event(name, event)))
+
+    @asyncio.coroutine
+    def handle_event(self, name, event):
+        """Coroutine to update state for a realtime event."""
+        import aioautomatic
+
+        # Fire a hass event
+        self.hass.bus.async_fire(EVENT_AUTOMATIC_UPDATE, event.data)
+
+        if event.vehicle.id not in self.vehicle_info:
+            # If vehicle hasn't been seen yet, request the detailed
+            # info for this vehicle.
+            _LOGGER.info("New vehicle found.")
+            try:
+                vehicle = yield from event.get_vehicle()
+            except aioautomatic.exceptions.AutomaticError as err:
+                _LOGGER.error(str(err))
+                return
+            yield from self.get_vehicle_info(vehicle)
+
+        kwargs = self.vehicle_info[event.vehicle.id]
+        if kwargs is None:
+            # Ignored device
+            return
+
+        # If this is a vehicle status report, update the fuel level
+        if name == "vehicle:status_report":
+            fuel_level = event.vehicle.fuel_level_percent
```

Comment:
This was tested and works for me. The `vehicle:status_report` uses the base realtime schema, and attaches a [vehicle](https://developer.automatic.com/api-reference/#vehicleserializer) object. Also, you linked to webhook discrepancies, but we're using websockets here.